### ISSUE TITLE
Remove support for network messages without a checksum

### ISSUE THREAD

**Issue by noescape00** [stratisproject/StratisBitcoinFullNode#3008]
`NetwokrPeerConnection.ReadMessageAsync` allows messages without a checksum which was introduced at version 60002. Min protocol version is 70000, therefore there is no need to support that.

**Comment by dangershony** [comment]
If I am not mistaken checksum is actually not needed anymore @NicolasDorier mentioned something to me the other day.

**Comment by NicolasDorier** [comment]
Yes, I removed it because it is quite useless while I needed to do some hacks to not allocate byte arrays.
